etc/generate-docs.sh

```diff
@@ -9,23 +9,53 @@ if ! command -v jazzy > /dev/null; then
   gem install jazzy || { echo "ERROR: Failed to locate or install jazzy; please install yourself with 'gem install jazzy' (you may need to use sudo)"; exit 1; }
 fi
 
+if ! command -v sourcekitten > /dev/null; then
+  gem install jazzy || { echo "ERROR: Failed to locate SourceKitten; please install yourself"; exit 1; }
+fi
+
 version=${1}
 
+# Ensure version is non-empty
+[ ! -z "${version}" ] || { echo "ERROR: Missing version string"; exit 1; }
+
 jazzy_args=(--clean
-            --author 'Matt Broadstone, Kaitlin Mahar, and Patrick Freed' 
-            --readme "docs/README.md" 
-            --author_url https://github.com/mongodb/mongo-swift-driver 
-            --github_url https://github.com/mongodb/mongo-swift-driver 
-            --theme fullwidth 
-            --documentation "Guides/*.md" 
             --github-file-prefix https://github.com/mongodb/mongo-swift-driver/tree/v${version} 
-            --module-version "${version}" 
-            --swift-build-tool spm)
+            --module-version "${version}")
+
+# Generate MongoSwift docs
+args=("${jazzy_args[@]}"  --output "docs-temp/MongoSwift" --module "MongoSwift" --config ".jazzy.yml" 
+        --root-url "https://mongodb.github.io/mongo-swift-driver/docs/MongoSwift/")
+jazzy "${args[@]}"
+
+# Generate MongoSwiftSync docs
+
+# we have to do some extra work to get re-exported symbols to show up
+sourcekitten doc --spm --module-name MongoSwift > mongoswift-docs.json
+python3 etc/filter_sourcekitten_output.py
+
+sourcekitten doc --spm --module-name MongoSwiftSync > mongoswiftsync-docs.json
+
+args=("${jazzy_args[@]}"  --output "docs-temp/MongoSwiftSync" --module "MongoSwiftSync" --config ".jazzy.yml" 
+        --sourcekitten-sourcefile mongoswift-filtered.json,mongoswiftsync-docs.json
+        --root-url "https://mongodb.github.io/mongo-swift-driver/docs/MongoSwiftSync/")
+jazzy "${args[@]}"
+
+rm mongoswift-docs.json
+rm mongoswift-filtered.json
+rm mongoswiftsync-docs.json
+
+git checkout gh-pages
+
+rm -rf docs/*
+cp -r docs-temp/* docs/
+rm -rf docs-temp
+
+echo '<html><head><meta http-equiv="refresh" content="0; url=MongoSwift/index.html" /></head></html>' > docs/index.html
+
+git add docs/
 
-modules=( MongoSwift MongoSwiftSync )
+git commit -m "${version} docs"
+git push
```

Comment:
what about leaving the push out of this script so we can test the docs out and see what they look like first? I think the release script might be more suited to actually pushing the docs.

Reply:
that's fair. I've refactored the two scripts such that all this one docs is generate the docs and put them into `docs-temp`. the release script now handles switching to the `gh-pages` branch, moving them into `docs`, committing and pushing them.